sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -97,11 +97,11 @@ class SparkSession private(
    * since that would cause every new session to reinvoke Spark Session Extensions on the currently
    * running extensions.
    */
-  private[sql] def this(sc: SparkContext) = {
+  private[sql] def this(sc: SparkContext, initialSessionOptions: Map[String, String]) = {
```

Comment:
Can we have another constructor: `private[sql] def this(sc: SparkContext) = this(sc, Map.empty)` instead?